util/llbutil/llbfactory/llb_factory.go

```diff
@@ -0,0 +1,106 @@
+package llbfactory
+
+import (
+	"sync"
+
+	"github.com/earthly/earthly/util/llbutil/pllb"
+
+	"github.com/moby/buildkit/client/llb"
+)
+
+var gSharedLocalMutex sync.Mutex
+
+var sharedLocalStateCache map[string]pllb.State
```

Comment:
I can see tying them together in a struct like:
```
type sharedLocalStateCache struct {
    mu sync.Mutex
    cache map[string]pllb.State
}
```

But I don't understand the benefit of including them in a different package. What's the use-case for accessing this cache from outside of the `LocalFactory.Construct()`'s function?

Reply:
I'm thinking to keep all the dependencies used by the Builder / Earthfile2llb self-contained and as much as possible not relying on any global object. So really.. I'm trying to avoid global objects as much as possible.

Reply:
In that case I think we should move the cache completely out of the factory code and into https://github.com/earthly/earthly/blob/local-include-pattern/earthfile2llb/adjust_copy_src.go#L53

Reply:
Got it, yeah that makes sense. I guess at that point it becomes an earthfile2llb-specific object.

Reply:
How does [this change](https://github.com/earthly/earthly/pull/1085/commits/b22a5906ba92ab2f828601ea9ad1b122d6a92a6d) look?

Reply:
Better... though it's still a global. I'd avoid sharing anything between builds. Today there's only one build per run, but later on we might have a service which runs a build per request or something like that. I think it's better if nothing is shared if we don't need to share across builds.

Reply:
That's a fair point, I wasn't thinking about the case of using the earthfile2llb package as a library that could persist between builds. I'll figure out a way to construct the cache when the build is triggered and pass it through.